### PR TITLE
Documentation rework

### PR DESCRIPTION
**Pull request by victorbenichoux**
In this PR I:

- Add some code example to the index, fix the header
- Add a logo to the docs
- Rework the Assets documentation
- Split the ModelLibrary documentation (there remains work here)

Again, the best way to test is to look at the documentation with `mkdocs serve`, focusing on `Developer guide/Model library/` and `Developer guide/Assets`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,26 +19,39 @@
...
-- **custom** `modelkit` is useful whenever you need to go beyond off-the-shelf models: custom processing, heuristics, business logic, different frameworks, etc.
-- **framework agnostic** you bring your own framework to the table, and you can use whatever code or library you want. Similarly, `modelkit` is not opinionated about how you build or train your models.
-- **organized** `modelkit` encourages you to version and share you ML library and artifacts with others, as a Python package or as a service. Let others use and evaluate your models!
-- **fast** `modelkit` add minimal overhead to prediction calls. Model predictions can be batched for speed (you define the batching logic).
-- **fast to code** Models only need to define their prediction logic and that's it. No cumbersome pre or postprocessing logic, branching options, etc... The boilerplate code is minimal and sensible.
-- **fast to deploy** Models can be served in a single CLI call using [fastapi](https://fastapi.tiangolo.com/)
+```python
+from modelkit import Model
+
+class MyModel(Model):
```
Comment (antoinejeannot) [inline README.md:29]: This quickstart example might be too simplistic to understand what modelkit really is about, and make anyone wants to use it. 

We could illustratre even more the features described below using the tutorial we already wrote ?

File: README.md [hunk 2]
```diff
@@ -19,26 +19,39 @@
...
 
 - **composable** Models can depend on other models, and evaluate them however you need to
 - **extensible** Models can rely on arbitrary supporting configurations files called _assets_ hosted on local or cloud object stores
 - **type-safe** Models' inputs and outputs can be validated by [pydantic](https://pydantic-docs.helpmanual.io/), you get type annotations for your predictions and can catch errors with static type analysis tools during development.
 - **async** Models support async and sync prediction functions. `modelkit` supports calling async code from sync code so you don't have to suffer from partially async code.
 - **testable** Models carry their own unit test cases, and unit testing fixtures are available for [pytest](https://docs.pytest.org/en/6.2.x/)
-- **robust** `modelkit` helps you follow software development best practices: all configurations and artifacts are explicitly versioned and tested.
+- **fast to deploy** Models can be served in a single CLI call using [fastapi](https://fastapi.tiangolo.com/)
+
+In addition, you will fined that `modelkit` is:
```
Comment (antoinejeannot) [inline README.md:46]:
```suggestion
In addition, you will find that `modelkit` is:
```

File: README.md [hunk 3]
```diff
@@ -19,26 +19,39 @@
 
 It is meant to bridge the gap between the different uses of your algorithms. With `modelkit` you can ensure that the same exact code will run in production, on your machine, or on data processing pipelines.
 
-## Features
+## Quickstart
 
-`modelkit`'s key features are:
+`modelkit` provides a straightforward and consistent way to wrap your prediction  code in a `Model` class:
```
Comment (antoinejeannot) [inline README.md:24]: We might want to chain the key features you talk about down below, and why not get inspired by the Requests Readme (which I love): https://docs.python-requests.org/en/master/

I thought of this, but it might be better up above:
> `modelkit` is a minimalist yet powerful MLOps library for Python, built for people who want to deploy ML models to production. It packs several key features which make your go-to-production journey a breeze:
// CODE EXAMPLE

File: README.md [hunk 4]
```diff
@@ -19,26 +19,39 @@
...
+class MyModel(Model):
+    def _predict(self, item):
+        # This is where your prediction logic goes, use any asset loaded by 
+        # _load here
+        ...
+        return result
+```
 
-And more:
+This will open the doors to all of `modelkit`'s cool features:
```
Comment (antoinejeannot) [inline README.md:37]:
```suggestion
This will open the doors to all of `modelkit`'s best features:
```

File: docs/assets/assets_dir.md [hunk 5]
```diff
@@ -1,42 +1,55 @@
...
 |   │   │   |   ├── content2
 |   │   │   |   |   ...
 |   │   │   ├── 0.1
 |   │   │   |   |   ...
 |    ...
 ```
 
-To retrieve the assets path, it is enough to run:
+!!! note
+    All previous versions of the assets are kept locally. It is save, however to delete local copies of the assets manually to save space.
```
Comment (antoinejeannot) [inline docs/assets/assets_dir.md:35]:
```suggestion
    All previous versions of the assets are kept locally. It is safe, however to delete local copies of the assets manually to save space.
```

File: docs/assets/assets_dir.md [hunk 6]
```diff
@@ -1,42 +1,55 @@
...
-mng.fetch_asset("some/asset:0.0") # will point to `some/asset/0.0`
-mng.fetch_asset("some/directory/asset:0.1") # will point to `some/asset/0.1`
+mng.fetch_asset("some/asset:0.0") # will point to `ASSETS_DIR/some/asset/0.0`
+mng.fetch_asset("some/directory/asset:0.1") # will point to `ASSETS_DIR/some/asset/0.1`
+mng.fetch_asset("some/directory/asset:0.1[content0]") # will point to `ASSETS_DIR/some/asset/0.1/content0`
 ```
+
+## Version resolution
+
+When an asset is request with the `version` not fully specified, `modelkit` may need to consult the remote storage to find the latest version. As a result, `modelkit`'s asset manager will in this context have a different behavior depending on whether a remote storage provider is parametrized:
```
Comment (antoinejeannot) [inline docs/assets/assets_dir.md:50]:
```suggestion
When an asset is requested with the `version` not fully specified, `modelkit` may need to consult the remote storage to find the latest version. As a result, `modelkit`'s asset manager will, in this context, have a different behavior depending on whether a remote storage provider is parametrized:
```

File: docs/assets/assets_dir.md [hunk 7]
```diff
@@ -1,42 +1,55 @@
...
+
+- **Without a remote store** find the latest version of the asset available in the `ASSETS_DIR`
+- **With a remote store** contact the remote store to find the latest version, see if it is present locally. If it is, use the local version, otherwise download the latest version.
```
Comment (antoinejeannot) [inline docs/assets/assets_dir.md:53]:
```suggestion
- **With a remote store** contact the remote store to find the latest version, check whether it is present locally. If it is, use the local version, otherwise download the latest version.
```

File: docs/assets/managing_assets.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+Assuming the asset is locally present at `asset_path` (either a file or a directory),
+create the remote asset `name:0.0` as follows:
+
+```python
+assets_store.new(asset_path, name)
+```
+
+!!! important
+    Creating new assets programmatically is likely not a very good idea.
```
Comment (antoinejeannot) [inline docs/assets/managing_assets.md:91]:
```suggestion
    Creating new assets programmatically is possible, even though it is not considered a good practice.
```

File: docs/assets/managing_assets.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+    Just do it manually once using the CLI.
```
Comment (antoinejeannot) [inline docs/assets/managing_assets.md:92]:
```suggestion
    Using the CLI is the prefered and safest way to manage assets.
```

File: docs/assets/remote_assets.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+```python
+class ModelWithAsset(Model):
+    CONFIGURATIONS = {
+        "model_with_asset": {"asset": "test/asset:1"} # meaning "version 1 of test/asset"
+    }
+```
+
+Assuming that you have parametrized a GCS object store, this will cause the `ModelLibrary` to:
+
+- download the objects at `gs://some-bucket-name/assets/test/1/yolo/*` locally (which storage provider and bucket is used depends on [your configuration](storage_provider.md))
```
Comment (antoinejeannot) [inline docs/assets/remote_assets.md:34]: the `which storage provider and bucket is used depends on` part is hard to follow

**Review by antoinejeannot — COMMENTED** [review]
I still have some files to review, but here's some feedback.
GG overall !

File: README.md (hunk 1 above)
Comment (victorbenichoux) [reply README.md:29]: I am trying to keep it as short as possible, but I do realise that it may well be overly simple. I wonder whether we should say something like:
"Now you can:
- expose it as an API with a single CLI
- use it in your code
- use it in batch
- share it"
Let's discuss this offline

**Review by antoinejeannot — APPROVED** [review]
